src/mono/mono/component/debugger-agent.c

```diff
@@ -8025,6 +8025,8 @@ type_commands_internal (int command, MonoClass *klass, MonoDomain *domain, guint
 			buffer_add_string (buf, f->name);
 			buffer_add_typeid (buf, domain, mono_class_from_mono_type_internal (f->type));
 			buffer_add_int (buf, f->type->attrs);
+			if (CHECK_PROTOCOL_VERSION(2, 61))
```

Comment:
What is the protocol version check for? Is the thread static field only expected for this version?

Reply:
It is checking if the client is prepared to receive this information. If the client version is older we cannot send this information otherwise it will break everything.
No, the thread static field already existed before, but it was detected in another way (calling other functions on runtime/reading metadata info), that I think it was much complex than this.